core/src/banking_stage.rs

```diff
@@ -1982,18 +1979,27 @@ impl BankingStage {
     fn receive_until(
         verified_receiver: &CrossbeamReceiver<Vec<PacketBatch>>,
         recv_timeout: Duration,
-        batching_timeout: Duration,
-        batch_size_upperbound: usize,
+        packet_count_upperbound: usize,
     ) -> Result<Vec<PacketBatch>, RecvTimeoutError> {
         let start = Instant::now();
         let mut packet_batches = verified_receiver.recv_timeout(recv_timeout)?;
```

Comment:
Do we need to do this initial receive, or is perf good enough if we just do the recv_timeout in the loop?

Reply:
If we do recv_timeout in the loop every receive attempt will wait for the timeout (if there are no more packet batches). The idea is that the first receive wait until timeout. If the receive is successful, try_recv as much as possible (guarded by the other upperbounds). `try_recv` is non blocking, so if there are no more packets to receive, it'll return immediately.

We have similar logic in `streamer.rs`